pkg/querier/queryrange/roundtrip.go

```diff
@@ -50,6 +58,7 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.DurationVar(&cfg.SplitQueriesByInterval, "querier.split-queries-by-interval", 0, "Split queries by an interval and execute in parallel, 0 disables it. You should use an a multiple of 24 hours (same as the storage bucketing scheme), to avoid queriers downloading and processing the same chunks. This also determines how cache keys are chosen when result caching is enabled")
 	f.BoolVar(&cfg.AlignQueriesWithStep, "querier.align-querier-with-step", false, "Mutate incoming queries to align their start and end with their step.")
 	f.BoolVar(&cfg.CacheResults, "querier.cache-results", false, "Cache query results.")
+	f.BoolVar(&cfg.SumShards, "querier.sum-shards", false, "Parse the ast and parallelize sums by shard.")
```

Comment:
Good call. Yes, there definitely are (I'm working on a cleaner impl for Loki because we control the `logql` pkg which should cover significantly more optimizations).